usr/lib/common/cert.c

```diff
@@ -41,17 +41,17 @@
 //
 CK_RV cert_check_required_attributes(TEMPLATE *tmpl, CK_ULONG mode)
 {
-    CK_ATTRIBUTE *attr = NULL;
-    CK_BBOOL found;
+    CK_ULONG val;
+    CK_RV rc;
 
     if (!tmpl)
         return CKR_FUNCTION_FAILED;
 
     if (mode == MODE_CREATE) {
-        found = template_attribute_find(tmpl, CKA_CERTIFICATE_TYPE, &attr);
-        if (found == FALSE) {
-            TRACE_ERROR("%s\n", ock_err(ERR_TEMPLATE_INCOMPLETE));
-            return CKR_TEMPLATE_INCOMPLETE;
+        rc = template_attribute_get_ulong(tmpl, CKA_CERTIFICATE_TYPE, &val);
+        if (rc != CKR_OK) {
+            TRACE_ERROR("Could not find CKA_CERTIFICATE_TYPE\n");
```

Comment:
Well, template_attribute_get_ulong already prints a trace message in case of CKR_ATTRIBUTE_VALUE_INVALID. So together with the trace message here it can clearly be determined if the attribute was not found (no trace message from template_attribute_get_ulong ) or was invalid.